frontend/src/generate-static-html.js

```diff
@@ -1,24 +1,42 @@
 
 import React from 'react';
 import ReactDOMServer from 'react-dom/server';
+import ReactMarkdown from 'react-markdown';
 
+import Loading from './app/components/Loading';
 
-function makeStaticString(headComponent, bodyComponent) {
+function makeStaticString(
+  prepareForClient, componentName, headComponent,
+  staticBodyComponent, dynamicBodyComponent, dynamicBodyArgument
+) {
   const head = ReactDOMServer.renderToStaticMarkup(headComponent);
-  const root = ReactDOMServer.renderToStaticMarkup(bodyComponent);
+  const staticRoot = ReactDOMServer.renderToStaticMarkup(staticBodyComponent);
+  let dynamicRoot;
+  let bootstrapScript;
+  if (prepareForClient) {
+    dynamicRoot = ReactDOMServer.renderToString(dynamicBodyComponent);
+    bootstrapScript = `<script src="/static/app/bootstrap.js"></script>
+<script>${componentName.slice(0, componentName.indexOf('.'))}.default(${dynamicBodyArgument});</script>`;
+  } else if (dynamicBodyComponent) {
+    dynamicRoot = ReactDOMServer.renderToStaticMarkup(dynamicBodyComponent);
+    bootstrapScript = '';
+  }
   return `<!DOCTYPE html>
 <html>
   ${head}
   <body class="blue">
     <div class="stars"></div>
-    <div id="page-container">${root}</div>
+    <div id="page-container">${dynamicRoot}</div>
+    <div id="static-root">${staticRoot}</div>
```

Comment:
Behavior of this static-root element is confusing me: I see that it's got the script includes for vendor JS and the individual page. But, it's also got a `<noscript>` block and the loading bars - both of which appear scrolled off the page because they're included *after* the server-rendered markup for the page content.

Maybe a) we don't need the loading bars after #2394; b) the noscript block should be at the top of the page; c) this div doesn't need to be removed?